unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2593,10 +2594,118 @@ bool includeInvisible
+                    start: new Vector3((float) p0.X, maxY, (float) p0.Y),
+                    end: new Vector3((float) p1.X, maxY, (float) p1.Y),
+                    color: Color.red,
+                    duration: 100.0f
+                );
+            }
+            #endif
+
+            return miconvexHull.Result.Select(p => ((float) p.X, (float) p.Y)).ToList();
+        }
+
+        /*
+        For each objectId, create a convex hull of the object from a top-down view.
+        The convex hull will be represented as a list of (x, z) world coordinates
+        such that the boundary formed by these coordinates forms the convex hull
+        of these points (smallest convex region enclosing the object's points).
+
+        If the objectIds (or objectTypes) parameter is non-null, then only objects with
+        those ids (or types) will be returned.
+        
+        ONLY ONE OF objectIds OR objectTypes IS ALLOWED TO BE NON-NULL.
+
+        Returns a dictionary mapping object ids to a list of (x,z) coordinates corresponding
+        to the convex hull of the corresponding object.
+        */
+        public void Get2DSemanticHulls(
+            List<string> objectIds = null,
+            List<string> objectTypes = null
+        ) {
+            if (objectIds != null && objectTypes != null) {
+                throw new ArgumentException(
+                    "Only one of objectIds and objectTypes can have a non-null value."
+                );
+            }
+
+            HashSet<string> allowedObjectTypesSet = null;
+            if (objectTypes != null) {
+                allowedObjectTypesSet = new HashSet<string>(objectTypes);
+            }
+
+            // Only consider sim objects which correspond to objectIds if given.
+            SimObjPhysics[] sopsFilteredByObjectIds = null;
+            if (objectIds != null) {
+                sopsFilteredByObjectIds = objectIds.Select(
+                    key => physicsSceneManager.ObjectIdToSimObjPhysics[key]
+                ).ToArray();
+            } else {
+                sopsFilteredByObjectIds = GameObject.FindObjectsOfType<SimObjPhysics>();
+            }
+
+            var objectIdToConvexHull = new Dictionary<string, List<(float x, float y)>>();
+            foreach (SimObjPhysics sop in sopsFilteredByObjectIds) {
+
+                // Skip objects that don't have one of the required types (if given)
+                if (
+                    allowedObjectTypesSet != null 
+                    && !allowedObjectTypesSet.Contains(sop.Type.ToString())
+                ) {
+                    continue;
+                }
+
+                #if UNITY_EDITOR
+                Debug.Log(sop.ObjectID);
+                #endif
+
+                objectIdToConvexHull[sop.ObjectID] = Get2DSemanticHull(sop.gameObject);
+            }
+            actionFinishedEmit(true, objectIdToConvexHull);
+        }
+
+        public void Get2DSemanticHulls(string objectId) {
```

Comment:
Wait, real quick: would rename this one to `Get2DSemanticHull` (no plural).

Reply:
I thought about this but I think I'd prefer to have it as it is as this way it's just an overload for a given action rather than having another whole action for this one behavior.

Reply:
This is true, for now. But, if we ever create a matching Python API, since Python does not support overloading methods, it is much more of a pain to get `Get2DSemanticHulls` working for `objectId`, `objectIds`, and `objectTypes` as input, rather than having it work for:
* `Get2DSemanticHulls`: `List[str]` (objectIds or objectTypes)
* `Get2DSemanticHulls`: `str`

Reply:
Fair enough, done.